sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/GlobalEndpointManager.java

```diff
@@ -159,6 +166,20 @@ public void close() {
         });
     }
 
+    public Mono<DatabaseAccount> getDatabaseAccountFromCache(URI defaultEndpoint) {
+        return this.databaseAccountAsyncCache.getAsync(StringUtils.EMPTY, null, () -> this.owner.getDatabaseAccountFromEndpoint(defaultEndpoint).single().doOnSuccess(databaseAccount -> {
+            if(databaseAccount != null) {
+                this.latestDatabaseAccount = databaseAccount;
+            }
+
+            this.refreshLocationAsync(databaseAccount, false);
```

Comment:
there is a bug here in `refreshLocationAsync(.)` invocation

Reply:
What bug , please explain . this is same as the code initially doing in RxDocumentClientImpl, via this.globalEndpointManager.refreshLocationAsync(databaseAccount, false).block();

Reply:
I think i see the bug will fix in next iteration

Reply:
Made refreshLocationAsync in async pipeline , added test cases to make sure this method is getting called